planty/posts/views.py

```diff
@@ -0,0 +1,50 @@
+from django.views.generic import CreateView, DetailView, UpdateView, DeleteView
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from .models import Post
+from django.urls import reverse_lazy
+
+
+class NewPostView(LoginRequiredMixin, CreateView):
+    # Template name has to be: model_form.html, shares template
+    # with update view
+    model = Post
+    fields = ['title', 'price', 'description', 'plant_image', 'plant_category']
+
+    # Model form views provide a form_valid() implementation that saves
+    # the model automatically.
+    def form_valid(self, form):
+        form.instance.user = self.request.user
+        return super().form_valid(form)
+
+
+class PostDetailView(DetailView):
+    # Template name has to be model_viewtype.html
```

Comment:
I can't see any template with `post_viewtype.html` name? Should this comment be updated?